lib/ruby_lsp/executor.rb

```diff
@@ -169,7 +169,18 @@ def folding_range(uri)
     sig { params(uri: String).returns(T::Array[Interface::CodeLens]) }
     def code_lens(uri)
       @store.cache_fetch(uri, "textDocument/codeLens") do |document|
-        Requests::CodeLens.new(document).run
+        document.parse
```

Comment:
Can we follow up with a PR to execute this request together with document symbol and document link? We can go through code lens on the same round of visits and use the caching mechanism to return the response.

Reply:
👍 https://github.com/Shopify/ruby-lsp/issues/679